bromelia/lib/etsi_3gpp_s6a/messages.py

```diff
@@ -707,4 +707,74 @@ def __init__(self,
                                  command_code=UPDATE_LOCATION_MESSAGE, 
                                  application_id=DIAMETER_APPLICATION_S6a_S6d)
 
-        DiameterRequest._load(self, locals())
\ No newline at end of file
+        DiameterRequest._load(self, locals())
+
+
+class InsertSubscriberDataRequest(DiameterRequest):
+    """Implementation of Insert-Subscriber-Data-Request (IDR) command as per 
+    clause 7.2.9 of ETSI TS 129 272 V15.8.0 (2019-07).
+
+    The Insert-Subscriber-Data-Request is indicated by the Command Code field set to 319 and the 'R' bit set in the Command Flags field.
+
+    Usage::
+
+        >>> from bromelia.lib.etsi_3gpp_s6a import IDR
+        >>> ulr_avps = {
+        ...     "destination_realm": "example.com",
+        ...     "user_name": "frodo",
+        ...     "visited_plmn_id": bytes.fromhex("ffffff")
+        ... }
+        >>> idr = IDR(**idr_avps)
+        >>> idr
+        <Diameter Message: 316 [ULR] REQ|PXY, 16777251 [3GPP S6a], 10 AVP(s)>
+    """    
+
+    mandatory = {
+                    "session_id": SessionIdAVP,
+                    "auth_session_state": AuthSessionStateAVP,
+                    "origin_host": OriginHostAVP,
+                    "origin_realm": OriginRealmAVP,
+                    "destination_realm": DestinationRealmAVP,
+                    "destination_host": DestinationHostAVP,
+                    "user_name": UserNameAVP,
+                    "subscription_data": SubscriptionDataAVP,
+                    "idr_flags": IdrFlagsAVP,
```

Comment:
IDR-Flags is not a mandatory AVP, so it can be moved to optionals class attribute